crate_universe/src/context.rs

```diff
@@ -272,147 +272,41 @@ impl Context {
-                            .deps
-                            .get_iter(conf)
-                            .expect("Lookup should be guaranteed")
+                IntoIterator::into_iter([
+                    &ctx.common_attrs.deps,
+                    &ctx.common_attrs.deps_dev,
+                    &ctx.common_attrs.proc_macro_deps,
+                    &ctx.common_attrs.proc_macro_deps_dev,
+                ])
+                .flat_map(|deps| {
+                    deps.configurations().into_iter().flat_map(move |conf| {
+                        deps.get_iter(conf).expect("Lookup should be guaranteed")
                     })
-                    .chain(
-                        ctx.common_attrs
-                            .deps_dev
-                            .configurations()
-                            .into_iter()
-                            .flat_map(move |conf| {
-                                ctx.common_attrs
-                                    .deps_dev
-                                    .get_iter(conf)
-                                    .expect("Lookup should be guaranteed")
-                            }),
-                    )
-                    .chain(
-                        ctx.common_attrs
-                            .proc_macro_deps
-                            .configurations()
-                            .into_iter()
-                            .flat_map(move |conf| {
-                                ctx.common_attrs
-                                    .proc_macro_deps
-                                    .get_iter(conf)
-                                    .expect("Lookup should be guaranteed")
-                            }),
-                    )
-                    .chain(
-                        ctx.common_attrs
-                            .proc_macro_deps_dev
-                            .configurations()
-                            .into_iter()
-                            .flat_map(move |conf| {
-                                ctx.common_attrs
-                                    .proc_macro_deps_dev
-                                    .get_iter(conf)
-                                    .expect("Lookup should be guaranteed")
-                            }),
-                    )
-            })
-            .cloned()
-            .collect();
-
-        // Search for any duplicate workspace member definitions
-        let duplicate_deps: Vec<CrateDependency> = workspace_member_dependencies
-            .iter()
-            .filter(|dep| {
-                workspace_member_dependencies
-                    .iter()
-                    .filter(|check| dep.id.name == check.id.name)
-                    .count()
-                    > 1
+                })
             })
-            .cloned()
-            .collect();
+            .collect()
+    }
 
-        workspace_member_dependencies
+    pub fn has_duplicate_workspace_member_dep(&self, dep: &CrateDependency) -> bool {
```

Comment:
Does this need to be public?

Reply:
It's called from src/rendering.rs which is a different module, so yes. I guess I could move the whole function to that module, or it's small enough to inline into the one place it's called. I put it here originally because this is where the equivalent logic for the tera template used to be (`flat_workspace_member_deps`).